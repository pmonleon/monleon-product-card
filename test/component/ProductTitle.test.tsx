import React from 'react';
import render  from 'react-test-renderer'
// import * as ReactDOM from 'react-dom';
import { ProductTitle, ProductCard } from '../../src/components';
import { products } from './data/products';

const mockProducts = products

describe('ProducTitle', () => { 

    test('Debe mostrar el componente Title con los dtaos de las props', () => { 
        const wrapper = render.create(
            <ProductTitle
                title='Custom Product'
                className='custom-class'
            />
        )
        
       // console.log(wrapper.toJSON())
        expect(wrapper.toJSON()).toMatchSnapshot()
        
        
    })
    test('Debe mostrar el componente Title con los datos del context', () => { 
        const wrapper2 = render.create(
            <ProductCard product={mockProducts[0]}>
                {
                    (  ) => (
                        <ProductTitle/>
                        )
                    }
            </ProductCard>
        )
        
        expect(wrapper2.toJSON()).toMatchSnapshot()
     })
})
