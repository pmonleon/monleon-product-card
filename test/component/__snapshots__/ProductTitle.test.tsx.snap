// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProducTitle Debe mostrar el componente Title con los datos del context 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card 1
  </span>
</div>
`;

exports[`ProducTitle Debe mostrar el componente Title con los dtaos de las props 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;
